database/engine/rrdengineapi.c

```diff
@@ -251,50 +308,66 @@ void rrdeng_store_metric_next(STORAGE_COLLECT_HANDLE *collection_handle,
                               SN_FLAGS flags)
 {
     struct rrdeng_collect_handle *handle = (struct rrdeng_collect_handle *)collection_handle;
-    struct rrdeng_metric_handle *metric_handle = (struct rrdeng_metric_handle *)handle->metric_handle;
-    struct rrdengine_instance *ctx = handle->ctx;
+    struct pg_cache_page_index *page_index = handle->page_index;
+    struct rrdengine_instance *ctx = handle->page_index->ctx;
     struct page_cache *pg_cache = &ctx->pg_cache;
     struct rrdeng_page_descr *descr = handle->descr;
-    RRDDIM *rd = metric_handle->rd;
 
     void *page;
     uint8_t must_flush_unaligned_page = 0, perfect_page_alignment = 0;
 
     if (descr) {
         /* Make alignment decisions */
 
-        if (descr->page_length == rd->rrdset->rrddim_page_alignment) {
+#ifdef NETDATA_INTERNAL_CHECKS
+        if(descr->end_time_ut + page_index->latest_update_every_s * USEC_PER_SEC != point_in_time_ut) {
+            char buffer[200];
```

Comment:
A minor suggestion. In normal situation in our code we set arrays like this with a size bigger than we give to `snprintfz`, for example, this [json](https://github.com/netdata/netdata/blob/294390f3b7514e8e23f3eb840be7eb25de442d6f/web/api/formatters/json/json.c#L20-L56) code, but here you are using the same size. I suggest to keep pattern and set here size `201`.

Reply:
done

Reply:
I am compiling your branch right now and I will run a last round of tests. :handshake: